kernel/src/process/unshare.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::sync::RwArc;
+
+use crate::{prelude::*, process::CloneFlags};
+
+/// Provides APIs for disassociating parts of the execution context
+pub trait ContextUnshareApi {
```

Comment:
This follows a different naming manner with `ContextNsAdminApi`.. Should we unify them?

Reply:
I think they should be `ContextUnshareAdminApi`/`ContextSetNsAdminApi` for now:

```rust
/// Provides administrative APIs for switching to new namespaces.
pub trait ContextSetNsAdminApi;

/// Provides administrative APIs for disassociating execution contexts.
pub trait ContextUnshareAdminApi;
```

If you choose `ContextNsAdminApi` (without **set**), I think `unshare_namespaces` should belong to it. But it is not the case.